18-references.Rmd

```diff
@@ -8,87 +8,97 @@
 '`
 
 ```{r include=FALSE, tidy=FALSE}
+# list cited packages
+pkgs <- c(
+  .packages(),
+  'animation',
+  'blastula',
+  'blogdown',
+  'bookdown',
+  'broom',
+  'Cairo',
+  'cairoDevice',
+  'condformat',
+  'dagitty',
+  'diagram',
+  'DiagrammeR',
+  'distill',
+  'downloadthis',
+  'DT',
+  'equatiomatic',
+  'ezknitr',
+  'flair',
+  'flexdashboard',
+  'flextable',
+  'formatR',
+  'formattable',
+  'gganimate',
+  'ggdag',
+  'ggplot2',
+  'gifski',
+  'googledrive',
+  'govdown',
+  'gt',
+  'gtsummary',
+  'here',
+  'huxtable',
+  'kableExtra',
+  'knitcitations',
+  'knitr',
+  'learnr',
+  'magick',
+  'nomnoml',
+  'officedown',
+  'officer',
+  'pagedown',
+  'pander',
+  'pixiedust',
+  'pkgdown',
+  'printr',
+  'r2d3',
+  'reactable',
+  'reticulate',
+  'revealjs',
+  'rgl',
+  'rhandsontable',
+  'rmarkdown',
+  'rmdformats',
+  'roxygen2',
+  'rsconnect',
+  'rticles',
+  'sass',
+  'spelling',
+  'stargazer',
+  'styler',
+  'svglite',
+  'tables',
+  'tangram',
+  'tikzDevice',
+  'tinytex',
+  'tufte',
+  'usethis',
+  'webshot',
+  'workflowr',
+  'xaringan',
+  'xfun',
+  'xtable',
+  'yaml',
+  'ztable'
+  # GH Only: 'equatiomatic', 'MonashEBSTemplates', 'plantuml', 'redoc', 'Statamarkdown', 'rmdrive'
+  # LaTeX: 'float', 'framed', 'tscolorbox', 'booktabs', 'longtable', 'flafter', 'subfig', 'fancyhdr', 'xcolor', 'listings', 'titling', 'animate'
+  # Bios only: 'shiny', 'leaflet', 'rolldown', 'crrri', 'projmgr', 'Rd2roxyen', 'servr', 'highr', 'mime', 'testit', 'fun', 'tufte'
+  # JavaScript: 'DataTables'
+)
+
+# install packages if missing
+lapply(pkgs, function(pkg) {
+    if (system.file(package = pkg) != '') return()
+    install.packages(pkg, repos = "http://cran.us.r-project.org") 
```

Comment:
```suggestion
    install.packages(pkg, repos = "https://cloud.r-project.org/") 
```

I would use this mirror as it is an automatic redirection to any mirror worlwide (see https://cran.r-project.org/mirrors.html)

Reply:
Another generic way for all code would be to set `repos` option in `index.Rmd`if non is already provided. This would allow to not override any set up CRAN repo. For example, it would help using RSPM in CI for example (with linux binaries available for Linux)